java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -29,83 +29,203 @@ public class CommandManager {
+     * @return A result promise of type T
+     */
+    public <T> CompletableFuture<T> submitNewCommand(
+            BaseTransaction transaction,
+            Optional<RequestRoutingConfiguration.Route> route,
+            RedisExceptionCheckedFunction<Response, T> responseHandler) {
+
+        RedisRequest.Builder command = prepareRedisRequest(transaction, route);
+        return submitNewCommand(command, responseHandler);
     }
 
     /**
-     * Build a protobuf command/transaction request object with routing options.<br>
-     * Used by {@link CommandManager}.
+     * Take a redis request and send to channel.
      *
-     * @param command Redis command type
-     * @param args Redis command arguments
+     * @param command The Redis command request as a builder to execute
+     * @param responseHandler The handler for the response object
+     * @return A result promise of type T
+     */
+    private <T> CompletableFuture<T> submitNewCommand(
+            RedisRequest.Builder command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
+        // write command request to channel
+        // when complete, convert the response to our expected type T using the given responseHandler
+        return channel.write(command, true).thenApplyAsync(responseHandler::apply);
+    }
+
+    /**
+     * Build a protobuf command request object with routing options.<br>
+     *
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
      * @param route Command routing parameters
      * @return An uncompleted request. {@link CallbackDispatcher} is responsible to complete it by
      *     adding a callback id.
      */
-    private RedisRequest.Builder prepareRedisRequest(
-            Command.RequestType command, String[] args, Optional<Route> route) {
+    protected RedisRequest.Builder prepareRedisRequest(
+            RequestType requestType,
+            String[] arguments,
+            Optional<RequestRoutingConfiguration.Route> route) {
         ArgsArray.Builder commandArgs = ArgsArray.newBuilder();
-        for (var arg : args) {
+        for (var arg : arguments) {
             commandArgs.addArgs(arg);
         }
 
         var builder =
                 RedisRequest.newBuilder()
                         .setSingleCommand(
-                                RedisRequestOuterClass.Command.newBuilder()
-                                        .setRequestType(mapRequestTypes(command))
+                                Command.newBuilder()
+                                        .setRequestType(mapRequestTypes(requestType))
                                         .setArgsArray(commandArgs.build())
                                         .build());
 
         if (route.isEmpty()) {
             return builder;
         }
 
-        if (route.get() instanceof SimpleRoute) {
+        if (route.get() instanceof RequestRoutingConfiguration.SimpleRoute) {
```

Comment:
and here too
It should show no diff for this piece of code (till the end of the function)

Reply:
fixed